synthetics-stats/src/handlers/common.ts

```diff
@@ -0,0 +1,28 @@
+import { ethereum } from "@graphprotocol/graph-ts";
+import { Transaction } from "../../generated/schema";
+
+export function getIdFromEvent(event: ethereum.Event): string {
+  return event.transaction.hash.toHexString() + ":" + event.logIndex.toString();
+}
+
+export function getOrCreateTransaction(event: ethereum.Event): Transaction {
+  let id = getIdFromEvent(event);
```

Comment:
i guess you can just use `event.transaction.hash.toHexString()` as id. otherwise for each handled event within transaction there will be separate Transaction